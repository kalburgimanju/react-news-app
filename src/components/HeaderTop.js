import React, { Component } from "react";
import { NavLink } from "react-router-dom";

class HeaderTop extends Component {
  renderNav() {
    const { isLogin } = this.props;
    if (isLogin) {
      return (
        <div className="container float_right">
          <NavLink to="/logout"> Logout </NavLink>
        </div>
      );
    }

    return (
      <div className="container float_right">
        <NavLink to="/login"> Login </NavLink>
        <NavLink to="/signup"> Sign Up </NavLink>
      </div>
    );
  }

  renderHello() {
    const { username, isLogin } = this.props;
    if (isLogin) {
      return (
        <div className="container float_left">
          Hello <NavLink to="/welcome"> {username} </NavLink>:)
        </div>
      );
    }
  }
  render() {
    return (
      <div className="menu__top">
        <div className="wrapper">
          <div className="container">
            {this.renderHello()}
            {this.renderNav()}
          </div>
        </div>
      </div>
    );
  }
}

export default HeaderTop;
