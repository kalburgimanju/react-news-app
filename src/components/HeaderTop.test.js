import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import HeaderTop from "./HeaderTop";

it("render login", () => {
  const username = "test@test.com";
  const wrapper = render(
    <Router>
      <HeaderTop isLogin={true} username={username} />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
  const text = wrapper.text();
  expect(text).toContain(username);
  expect(text).toContain("Logout");
  expect(text).not.toContain("Login");
});

it("render not login", () => {
  const username = "test@test.com";
  const wrapper = render(
    <Router>
      <HeaderTop isLogin={false} username={username} />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
  const text = wrapper.text();
  expect(text).not.toContain(username);
  expect(text).not.toContain("Logout");
  expect(text).toContain("Login");
  expect(text).toContain("Sign Up");
});
