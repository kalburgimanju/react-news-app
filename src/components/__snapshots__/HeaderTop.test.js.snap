// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render login 1`] = `
<div
  class="menu__top"
>
  <div
    class="wrapper"
  >
    <div
      class="container"
    >
      <div
        class="container float_left"
      >
        Hello 
        <a
          aria-current="false"
          href="/welcome"
        >
           test@test.com 
        </a>
        :)
      </div>
      <div
        class="container float_right"
      >
        <a
          aria-current="false"
          href="/logout"
        >
           Logout 
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`render not login 1`] = `
<div
  class="menu__top"
>
  <div
    class="wrapper"
  >
    <div
      class="container"
    >
      <div
        class="container float_right"
      >
        <a
          aria-current="false"
          href="/login"
        >
           Login 
        </a>
        <a
          aria-current="false"
          href="/signup"
        >
           Sign Up 
        </a>
      </div>
    </div>
  </div>
</div>
`;
